eo-maven-plugin/src/test/java/org/eolang/maven/FakeMaven.java

```diff
@@ -181,16 +186,41 @@ public Map<String, Path> result() throws IOException {
         );
     }
 
+    /**
+     * Suffix for the program name or path.
+     * - main_1.eo
+     * - foo.x.main100
+     * - main.eo
+     * @param index Number of the program.
+     * @return String suffix.
+     */
+    static String suffix(final int index) {
+        final String suffix;
+        if (index == 0) {
+            suffix = "";
+        } else {
+            suffix = String.format("_%d", index);
+        }
+        return suffix;
+    }
+
     /**
      * Adds eo program to a workspace.
-     * @param path Relative path where to save EO program
      * @param content EO program content.
      * @return The same maven instance.
      * @throws IOException If method can't save eo program to the workspace.
      */
-    private FakeMaven withProgram(final Path path, final String content) throws IOException {
+    private FakeMaven withProgram(final String content) throws IOException {
+        final Path path = Paths.get(
+            String.format("foo/x/main%s.eo", FakeMaven.suffix(this.current.get()))
```

Comment:
@volodya-lombrozo with concurrent execution we still can get the same number here.
We also don't have any synchronization between `concurrent.get()` and `concurrent.incrementAndGet()` so some other thread might increment the value and foreign catalog might get wrong file written.

Reply:
@mximp `FakeMaven` isn't thread safe for now and it doesn't have to be. The idea is to create single `FakeMaven` and work with it. I can't imagine the situation where I have to prepare environment in parallel. 
If you insist, I can create a puzzle, but I don't see the reason for it. What do you think?

Reply:
@volodya-lombrozo Can we mark `FakeMave` as not thread-safe in JavaDoc please

Reply:
@mximp Done